src/coreclr/src/jit/lowerxarch.cpp

```diff
@@ -1081,6 +1092,936 @@ void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
+        {
+            // These intrinsics are meant to set the same value to every element
+            // so we'll just specially handle it here and copy it into the remaining
+            // indices.
+
+            for (unsigned i = 1; i < simdSize / genTypeSize(baseType); i++)
+            {
+                HandleArgForHWIntrinsicCreate(op1, i, vecCns, baseType);
+            }
+        }
+    }
+
+    if (argCnt == cnsArgCnt)
+    {
+        if (op1->OperIsList())
+        {
+            for (argList = op1->AsArgList(); argList != nullptr; argList = argList->Rest())
+            {
+                BlockRange().Remove(argList->Current());
+            }
+        }
+        else
+        {
+            BlockRange().Remove(op1);
+
+            if (op2 != nullptr)
+            {
+                BlockRange().Remove(op2);
+            }
+        }
+
+        CORINFO_FIELD_HANDLE hnd = comp->GetEmitter()->emitAnyConst(&vecCns, simdSize, emitDataAlignment::Required);
+        GenTree* clsVarAddr      = new (comp, GT_CLS_VAR_ADDR) GenTreeClsVar(GT_CLS_VAR_ADDR, TYP_I_IMPL, hnd, nullptr);
+        BlockRange().InsertBefore(node, clsVarAddr);
+
+        node->ChangeOper(GT_IND);
+        node->gtOp1 = clsVarAddr;
+
+        // TODO-XARCH-CQ: We should be able to modify at least the paths that use Insert to trivially support partial
+        // vector constants. With this, we can create a constant if say 50% of the inputs are also constant and just
+        // insert the non-constant values which should still allow some gains.
+
+        return;
+    }
+    else if (argCnt == 1)
+    {
+        // We have the following:
+        //
+        // /--*  <value>
+        // *  HWINTRINSIC simd16 T Create
+
+        if (intrinsicId == NI_Vector256_Create)
+        {
+            if (comp->compOpportunisticallyDependsOn(InstructionSet_AVX2))
+            {
+                // We will be constructing the following parts:
+                //
+                //      /--*  <value>
+                // t1 = *  HWINTRINSIC simd16 ubyte CreateScalarUnsafe
+                //      /--*  t1     simd16
+                // t0 = *  HWINTRINSIC simd32 T BroadcastScalarToVector256
+
+                op1 = comp->gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_Vector128_CreateScalarUnsafe, baseType, 16);
+                BlockRange().InsertBefore(node, op1);
+                LowerNode(op1);
+
+                node->gtOp1 = op1;
+                node->gtOp2 = nullptr;
+
+                node->gtHWIntrinsicId = NI_AVX2_BroadcastScalarToVector256;
+                return;
+            }
+
+            assert(comp->compIsaSupportedDebugOnly(InstructionSet_AVX));
+
+            // We will be constructing the following parts:
+            //
+            // t2 =    CNS_INT   int    0
+            //      /--*  t?     simd32
+            //      +--*  t?     simd16
+            //      +--*  t?     int
+            // t0 = *  HWINTRINSIC simd32 T InsertVector128
```

Comment:
This comment doesn't match what's generated below (e.g. you're creating a constant of 0x01). Also, it would be easier to follow if you referred to 'op1', 'op2' and 'idx' here (and maybe add 'v' for the target of `NI_Vector128_Create` and `result` for the final value, instead of using 't2' and 't?'.
I would also do something similar for the comments below.

Reply:
Looks like I forgot to fill in the generated LIR for this one. I'm updating it and the other comments to be closer to the following:
```csharp
            // We will be constructing the following parts:
            //          /--*  op1  T
            //   tmp  = *  HWINTRINSIC simd16 T Create
            //          /--*  tmp  simd16
            //          *  STORE_LCL_VAR simd16
            //   tmp  =    LCL_VAR simd16
            //   dup  =    LCL_VAR simd16
            //          /--*  dup  simd16
            //   dup  = *  HWINTRINSIC simd16 T ToVector256Unsafe
            //   idx  =    CNS_INT     int    0
            //          /--*  dup  simd32
            //          +--*  tmp  simd16
            //          +--*  idx  int
            //   node = *  HWINTRINSIC simd32 T InsertVector128

            // This is roughly the following managed code:
            //   var tmp = Vector128.Create(op1);
            //   var dup = tmp.ToVector256Unsafe(tmp);
            //   return Avx2.BroadcastScalarToVector256(dup, tmp, 1);
```